certbot/plugins/common.py

```diff
@@ -99,7 +100,131 @@ def dest(self, var):
     def conf(self, var):
         """Find a configuration value for variable ``var``."""
         return getattr(self.config, self.dest(var))
-# other
+
+
+class PluginStorage(object):
+    """Class implementing storage functionality for plugins"""
+
+    def __init__(self, config, classkey):
+        """Initializes PluginStorage object storing required configuration
+        options.
+
+        :param .configuration.NamespaceConfig config: Configuration object
+        :param str classkey: class name to use as root key in storage file
+
+        :returns: Plugin storage object
+        :rtype: dict
+
+        :raises .errors.PluginStorageError: when unable to open or read the file
+        """
+
+        self.config = config
+        self.classkey = classkey
+        self.initialized = False
+        self._data = None
+        self.storagepath = None
+
+    def initialize_storage(self):
+        """Initializes PluginStorage data and reads current state from the disk
+        if the storage json exists."""
+
+        if hasattr(self.config, "config_dir") and os.path.isdir(self.config.config_dir):
+            self.storagepath = os.path.join(self.config.config_dir, ".pluginstorage.json")
+            self._data = self.load()
+        else:
+            self.storagepath = None
+        self.initialized = True
+
+    def load(self):
```

Comment:
Like Erica said, I think we should consider making this private, especially because modifications to the returned dictionary won't be saved and you have to use `put`, `fetch`, and `save`.

Reply:
If you don't want to make it private, please explain why, but I would  make this method return nothing and instead just set `self._data`. If you keep it private, I have no preference about whether or not you make this change.